Build process: new npm module that allows gulp-uglify to process es6

dereksmart opened the pull request:
#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

Adds a new npm module [uglify-es](https://www.npmjs.com/package/uglify-es) that allows `gulp-uglify` to process es6. 

It was broken when we [updated](https://github.com/Automattic/jetpack/pull/10849) to gulp-uglify 3.0, as it now depends on the default uglifyJS within it.

This is now the [suggested method](https://github.com/terinjokes/gulp-uglify#using-a-different-uglifyjs) for es6 via gulp-uglify.

#### Testing instructions:
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

- `yarn install`
- run `yarn build-production-client`
- Verify no errors
- Check `/_inc/build/admin.js` file. It should be minified.

run the dev commands as well: 
- `yarn build`
- `yarn watch`
- verify that nothing is broken in the build, and the admin page renders fine. 

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->

none

dereksmart committed:
New npm module for allowing gulp-uglify to process es6

dereksmart committed:
remove debug code

brbrr reviewed (dismissed):
It builds well, although it looks like these changes kinda broke WPCOM Masterbar styling

<img width="1362" alt="monosnap 2019-01-04 10-43-26" src="https://user-images.githubusercontent.com/5654161/50673078-42558c00-100e-11e9-995f-26f45e92c000.png">

jeherve committed:
Move away from uglify-es. Use terser instead.

uglify-es is not maintained anymore:
https://github.com/mishoo/UglifyJS2/issues/3156#issuecomment-392943058
Terser seems to be the replacement of choice, and is also what we use in Calypso:
https://github.com/Automattic/wp-calypso/pull/26531

This commit introduces gulp-minify, which uses terser behind the scenes:
https://www.npmjs.com/package/gulp-minify

jeherve committed:
Update yarn.lock

jeherve commented:
I've made some changes to this PR:

uglify-es is not maintained anymore:
https://github.com/mishoo/UglifyJS2/issues/3156#issuecomment-392943058
Terser seems to be the replacement of choice, and is also what we use in Calypso:
https://github.com/Automattic/wp-calypso/pull/26531

This commit introduces gulp-minify, which uses terser behind the scenes:
https://www.npmjs.com/package/gulp-minify

I've tested `yarn build` and `yarn build-production-client`, and in both cases the Jetpack dashboard appears to work.

brbrr reviewed (approved):
The issue I mentioned above seems like not related to uglification. Which means - :shipit:
